### PR TITLE
INC-120 Provide single location lookup by description (unique)

### PR DESCRIPTION
**Pull request by Mjwillis**

### CODE REVIEW

File: src/main/java/uk/gov/justice/hmpps/prison/service/LocationService.java [hunk 1]
```diff
@@ -100,6 +100,24 @@ public Location getLocation(final long locationId, boolean includeInactive) {
             .orElseThrow(EntityNotFoundException.withId(locationId));
     }
 
+    public Optional<Location> getLocationByDescription(final String description) {
+        return agencyInternalLocationRepository.findOneByDescription(description)
+            .map(
+                l -> LocationProcessor.processLocation(Location.builder()
```
Comment (steverendell) [inline src/main/java/uk/gov/justice/hmpps/prison/service/LocationService.java:106]: Not too sure about the parameter name !

Comment (Mjwillis) [reply src/main/java/uk/gov/justice/hmpps/prison/service/LocationService.java:106]: I know, but that is what it is called in the DB and the repo

Comment (Mjwillis) [reply src/main/java/uk/gov/justice/hmpps/prison/service/LocationService.java:106]: @steverendell what do you think ?  code, id, locationCode, Basically it is the description field that happens to be unique

Comment (steverendell) [reply src/main/java/uk/gov/justice/hmpps/prison/service/LocationService.java:106]: Anything like code or location is fine, its just that an 'l' looks like a '1' .. and is hard to spot in the block too 🙂

**Review by steverendell — APPROVED** [review]
lgtm

**Pull request merged**